analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -352,13 +352,75 @@ public NormalClass NormalClassSetToNull()
 // https://github.com/SonarSource/sonar-dotnet/issues/4015
 public class Repro_4015
 {
-    public void Method()
+    Expression<Func<string>> fieldExpression;
+    Func<string> fieldFunc;
+
+    public void Compliant_SimpleLambda()
     {
-        string localVariable = null;    // Noncompliant FP, changing the Expression<Func<T>> below changes the behavior of the code
+        string localVariable = null;    // Compliant, changing the Expression<Func<T>> below changes the behavior of the code
+        WithExpressionArgument(x => localVariable);
+    }
 
+    public void Compliant_ParenthesizedLambda()
+    {
+        string localVariable = null;    // Compliant, changing the Expression<Func<T>> below changes the behavior of the code
         WithExpressionArgument(() => localVariable);
     }
 
+    public void Compliant_ParenthesizedExpression()
+    {
+        string localVariable = null;
+        WithExpressionArgument(((() => localVariable)));
+    }
+
+    public void Noncompliant_Func()
+    {
+        string localVariable = null;    // Noncompliant, this can be const
+        WithFuncArgument(() => localVariable);
+    }
+
+    public void Compliant_Assignment()
+    {
+        string a = null;
+        string b = null;
+        string c = null;
+        string d = null;
+        var repro = new Repro_4015();
+        Expression<Func<string>> variableExpression;
+
+        // Compliant cases, changing the Expression<Func<T>> below changes the behavior of the code
+        fieldExpression = () => a;
+        WithExpressionArgument(fieldExpression);
+
+        repro.fieldExpression = () => b;
+        WithExpressionArgument(repro.fieldExpression);
+
+        variableExpression = () => c;
+        WithExpressionArgument(variableExpression);
+
+        variableExpression = ((() => d));
+        WithExpressionArgument(variableExpression);
+    }
+
+    public void Noncompliant_Assignment()
+    {
+        string a = null;    // Noncompliant
+        string b = null;    // Noncompliant
+        string c = null;    // Noncompliant
+        var repro = new Repro_4015();
+        Func<string> variableFunc;
+
+        // Compliant cases, changing the Expression<Func<T>> below changes the behavior of the code
```

Comment:
The cases are marked as compliant but the variables are noncompliant. Did you forgot to remove the comment?